### PR TITLE
feat(provider/ecs): Added IAM role reader to ECS

### PR DESCRIPTION
**Pull request by BrunoCarrier**
feat(provider/ecs): Added IAM role reader to ECS

@anotherchrisberry @jrsquared @christopherthielen

### CODE REVIEW

File: app/scripts/modules/ecs/iamRoles/IRole.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface IRoleDescriptor {
+  name: string;
+  trustRelationships: string[];
+  id: string;
+  accountName: string;
+}
+
```
Comment (anotherchrisberry) [inline app/scripts/modules/ecs/iamRoles/IRole.ts]: extra whitespace here

File: app/scripts/modules/ecs/iamRoles/iamRole.read.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { API_SERVICE, Api } from 'core/api/api.service';
+import { IRoleDescriptor } from './IRole';
+
+export class IamRoleReader {
+  public constructor(private API: Api) { 'ngInject'; }
+
+  public listRoles(provider: string): IPromise<IRoleDescriptor[]> {
+    return this.API
+      .all('roles')
+      .all(provider)
```
Comment (anotherchrisberry) [inline app/scripts/modules/ecs/iamRoles/iamRole.read.service.ts]: Does `provider` need to be parameterized? If so, this probably belongs in the core module, not the ECS module.

Comment (BrunoCarrier) [reply app/scripts/modules/ecs/iamRoles/iamRole.read.service.ts]: Mmmm @robzienert are roles to be a spinnaker primitive, or not?  If I remember well from a PR comment on Gate, you wanted them to be ECS specific.  What's your recommended approach here?

File: app/scripts/modules/ecs/iamRoles/IRole.ts (hunk 1 above)
Comment (BrunoCarrier) [reply app/scripts/modules/ecs/iamRoles/IRole.ts]: Removed an extra white line

**Pull request merged**